### ISSUE TITLE
Refactoring the mp-intro guide

### ISSUE THREAD

**Issue by evelinec** [OpenLiberty/guide-microprofile-intro#35]
Main problems
Some of the links have not been updated to reflect the code changes. For example, the guide still mentions the endpoint systems which we used in the original draft of the code, but it has been replaced by hosts. There was also a link pointing to inventory/properties which is completely invalid since it doesn't actually serve anything. Some minor issues as well like not being clear what files should be created, explanations in wrong parts, etc.

Other problems
The guide doesn't really flow that well. Each section feels like a new chapter. There are also a lot of unclear sentences that definitely need to be rewritten to get their point across better. The guide is also quite long and can probably be refactored to be less shorter.

**Comment by yeekangc** [comment]
@Kubik42 @JunyiSun, please summarize the issues that we are trying to address here so we can ensure focus.

**Comment by Kubik42** [comment]
**Main problems**
Some of the links have not been updated to reflect the code changes. For example, the guide still mentions the endpoint `systems` which we used in the original draft of the code, but it has been replaced by `hosts`. There was also a link pointing to `inventory/properties` which is completely invalid since it doesn't actually serve anything. Some minor issues as well like not being clear what files should be created, explanations in wrong parts, etc.

**Other problems**
The guide doesn't really flow that well. Each section feels like a new chapter. There are also a lot of unclear sentences that definitely need to be rewritten to get their point across better. The guide is also quite long and can probably be refactored to be less shorter.
